src/components/datatable.component.ts

```diff
@@ -450,7 +450,7 @@ export class DatatableComponent implements OnInit, AfterViewInit {
     });
 
     this.recalculateColumns(cols, idx);
-    this.columns = cols;
+    this._columns = cols;
```

Comment:
We need this because setting property will recalculateColumns again. We just have calculated columns so there is no need to recalculate I guess